erpnext/stock/doctype/batch/batch.py

```diff
@@ -120,6 +110,28 @@ def item_has_batch_enabled(self):
 		if frappe.db.get_value("Item", self.item, "has_batch_no") == 0:
 			frappe.throw(_("The selected item cannot have Batch"))
 
+	def before_save(self):
+		has_expiry_date, shelf_life_in_days = frappe.db.get_value('Item', self.item, ['has_expiry_date', 'shelf_life_in_days'])
+		if not self.expiry_date and has_expiry_date and shelf_life_in_days:
+			self.expiry_date = add_days(self.manufacturing_date, shelf_life_in_days)
+
+		if has_expiry_date and not self.expiry_date:
+			frappe.throw('Expiry date is mandatory for selected item')
+			frappe.msgprint('Set items shelf life in days, to set expiry based on manufacturing_date plus self life ')
+
+	def get_name_from_naming_series(self):
```

Comment:
`get_name_from_naming_series` was actual written away from the object so it can be reusable. Is there any other reason why you decided to turn it to a method?

Reply:
that is because, i have added support for jinja, user can refer to any attribute it would be available in self. that is why added to class method.

Reply:
cool